src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -689,49 +850,198 @@ private static IEnumerable<double> LazySequence(double records, double start, do
             }
         }
 
-        public static FormulaValue Abs(IRContext irContext, NumberValue[] args)
+        private static IEnumerable<decimal> LazySequenceDecimal(double records, decimal start, decimal step)
         {
-            var arg0 = args[0];
-            var x = arg0.Value;
-            var val = Math.Abs(x);
-            return new NumberValue(irContext, val);
+            var x = start;
+            for (var i = 1; i <= records; i++)
+            {
+                yield return x;
+                x += step;
+            }
         }
 
-        public static FormulaValue Round(IRContext irContext, NumberValue[] args)
+        public static FormulaValue Abs(IRContext irContext, FormulaValue[] args)
         {
-            var numberArg = args[0].Value;
-            var digitsArg = args[1].Value;
+            var arg0 = args[0];
+
+            if (arg0 is NumberValue num)

```

Comment:
> if (arg0 is NumberValue num) [](http://example.com/codeflow?start=11&length=29)

This kind of checking motivates me to use overloads.

Reply:
Discussed at length today.  Seeing if you come up with some type system magic to help with validating the return value.  If not, we can check in this way, and look for a better option for that part later.  I have cleaned up the separation between decimal and float values and enhanced the checking of the integer arguments for Round*, Trunc, and Sequence.

Reply:
AbsFloat/AbsDecimal is a good mitigation